charts/telepresence/templates/certificate.yaml

```diff
@@ -0,0 +1,32 @@
+{{- if eq .Values.agentInjector.certificate.method "certmanager" }}
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ .Values.agentInjector.secret.name }}
+spec:
+  secretName: {{ .Values.agentInjector.secret.name }}
+  commonName: {{ .Values.agentInjector.name }}
+  dnsNames:
+  - {{ (printf "%s.%s" .Values.agentInjector.name .Release.Namespace ) }}
+  - {{ (printf "%s.%s.svc" .Values.agentInjector.name .Release.Namespace ) }}
+  {{- with .Values.agentInjector.certificate.certmanager.duration }}
```

Comment:
Besides renaming `issuer` to `issuerRef` and tweaking the `duration` format, the fields below are just copied from an otherwise anonymous map `.Values.agentInjector.certificate.certmanager`. This raises some questions:
1. Why not start all this with a `{{- with .Values.agentInjector.certificate.certmanager }}` and avoid repeating that over and over?
2. Why is the original duration format different and why using `issuer` instead of `issurerRef`?

I guess the main question is, why not just expand the whole certmanager map here and require that it uses a correct duration format and names to start with?

Reply:
You did bring an interesting point. I like it. I did submit another commit regarding this. My previous commit has been done with inspiration on the Cilium and ArgoCd Helm chart.